ldap_jwt_auth/routers/verify.py

```diff
@@ -0,0 +1,31 @@
+"""
+Module for providing an API router which defines a route for managing verification of JWT tokens.
+"""
+import logging
+from typing import Annotated
+
+from fastapi import APIRouter, Body, Depends, HTTPException, Response, status
+
+from ldap_jwt_auth.auth.jwt_hanlder import JWTHandler
+from ldap_jwt_auth.core.exceptions import InvalidJWTError
+
+logger = logging.getLogger()
+
+router = APIRouter(prefix="/verify", tags=["authentication"])
+
+
+@router.post(
+    path="/",
+    summary="Verify that a JWT token was generated by this authentication service",
+    response_description="200 status code (no response body) if the token is valid",
+)
+def verify_token(
+    jwt_handler: Annotated[JWTHandler, Depends(JWTHandler)],
+    token: Annotated[str, Body(description="The JWT token to verify")],
+) -> Response:
+    # pylint: disable=missing-function-docstring
+    try:
+        jwt_handler.verify_token(token)
+        return Response(status_code=status.HTTP_200_OK)
+    except InvalidJWTError as exc:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Invalid JWT token provided") from exc
```

Comment:
```suggestion
def verify_token(
    jwt_handler: Annotated[JWTHandler, Depends(JWTHandler)],
    token: Annotated[str, Body(description="The JWT token to verify")],
    status_code=status.HTTP_200_OK
) -> Response:
    # pylint: disable=missing-function-docstring
    try:
        jwt_handler.verify_token(token)
    except InvalidJWTError as exc:
        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Invalid JWT token provided") from exc
```

Reply:
Doing what you suggested returns a body that includes `200` which is expected. Maybe `204` should instead be returned given nothing is returned by this endpoint. `HTTP_204_NO_CONTENT` is used for responses without a body but then that would require changes in SciGateway.

Reply:
Yes I agree it should really be 204, it would need changing on the react 18 branch which would also effect operations gateway api which is why I didn't mention it at the time but it should be a minor change.

Reply:
https://github.com/ral-facilities/operationsgateway-api too needs to be updated.

Reply:
Will be addressed separately as part of https://github.com/ral-facilities/ldap-jwt-auth/issues/17